lib/src/merge.rs

```diff
@@ -249,25 +249,62 @@ impl<T> Merge<T> {
         }
     }
 
-    /// Simplify the merge by joining diffs like A->B and B->C into A->C.
-    /// Also drops trivial diffs like A->A.
-    pub fn simplify(mut self) -> Self
+    /// Returns a vector mapping of a value's index in the simplified merge to
+    /// its original index in the unsimplified merge.
+    ///
+    /// The merge is simplified by removing identical values in add and remove
+    /// values.
+    fn get_simplified_mapping(&self) -> Vec<usize>
     where
         T: PartialEq,
     {
+        let unsimplified_len = self.values.len();
+        let mut original_indices = (0..unsimplified_len).collect_vec();
```

Comment:
Minor, optional: I would call this `original_index`, `original_index[5]` reads better to me.

Reply:
It could also be `simplified_to_original_indices`.

What makes me slightly confused, and what I wanted to improve, is that the array called `original_indicies` does not contain all the original indicies.

I do not feel strongly about this.